front/app/containers/ProjectsShowPage/shared/header/ProjectActionButtons.tsx

```diff
@@ -37,147 +39,198 @@ import { selectPhase } from 'containers/ProjectsShowPage/timeline/events';
-        currentPhase.attributes.end_at,
-      ]) === 'past'
-    : false;
-  const inputTerm = getInputTerm(
-    project.attributes.process_type,
-    project,
-    phases
-  );
-
-  return (
-    <Container className={className || ''}>
-      {((process_type === 'continuous' &&
-        participation_method === 'ideation') ||
-        currentPhase?.attributes.participation_method === 'ideation') &&
-        isNumber(ideas_count) &&
-        ideas_count > 0 && (
-          <SeeIdeasButton
-            id="e2e-project-see-ideas-button"
-            buttonStyle="secondary"
-            onClick={scrollTo('project-ideas')}
-            fontWeight="500"
-          >
-            <FormattedMessage
-              {...getInputTermMessage(inputTerm, {
-                idea: messages.seeTheIdeas,
-                option: messages.seeTheOptions,
-                project: messages.seeTheProjects,
-                question: messages.seeTheQuestions,
-                issue: messages.seeTheIssues,
-                contribution: messages.seeTheContributions,
-              })}
+    const { enabled, disabledReason } = getSurveyTakingRules({
+      project,
+      phaseContext: currentPhase,
+      signedIn: !isNilOrError(authUser),
+    });
+    const registrationNotCompleted =
+      !isNilOrError(authUser) && !authUser.attributes.registration_completed_at;
+    const shouldVerify = !!(
+      disabledReason === 'maybeNotVerified' || disabledReason === 'notVerified'
+    );
+
+    // Using the same rules used to show the sign wrapper in survey display
+    const showSignIn =
+      shouldVerify ||
+      disabledReason === 'maybeNotPermitted' ||
+      registrationNotCompleted;
+
+    const handleTakeSurveyClick = (event: FormEvent) => {
+      if (showSignIn) {
+        openSignUpInModal({
+          flow: 'signup',
+          verification: shouldVerify,
+          verificationContext: undefined,
+          action: () => scrollTo('project-survey')(event),
+        });
+      }
+
+      if (enabled === true) {
+        scrollTo('project-survey')(event);
+      }
+    };
+
+    const { process_type, participation_method, publication_status } =
+      project.attributes;
+    const ideas_count =
+      process_type === 'continuous'
+        ? project.attributes.ideas_count
+        : currentPhase?.attributes.ideas_count;
+    const hasProjectEnded = currentPhase
+      ? pastPresentOrFuture([
+          currentPhase.attributes.start_at,
+          currentPhase.attributes.end_at,
+        ]) === 'past'
+      : false;
+    const inputTerm = getInputTerm(
+      project.attributes.process_type,
+      project,
+      phases
+    );
+
+    return (
+      <Container className={className || ''}>
+        {((process_type === 'continuous' &&
+          participation_method === 'ideation') ||
+          currentPhase?.attributes.participation_method === 'ideation') &&
+          isNumber(ideas_count) &&
+          ideas_count > 0 && (
```

Comment:
Not blocking (just a style suggestion), but maybe these complex conditions that determine which buttons to show can be extracted into variables, e.g.

```
const showSeeIdeasButton = (
  process_type === 'continuous' &&
  participation_method === 'ideation'
) || (
  currentPhase?.attributes.participation_method === 'ideation') &&
  isNumber(ideas_count) &&
  ideas_count > 0
);
```